dateValidator.js

```diff
@@ -0,0 +1,15 @@
+function dateValidator(date) {
+    if (date == undefined) {
```

Comment:
Recomendo validar usando regex. Exemplos que dariam um resultado diferente do esperado:

1. const date = ""
2. const date = "error"
3. const date = 1010
4. const date = {}

```suggestion
    if (!/^\d{4}\-\d{1,2}\-\d{1,2}$/.test(date)) {
```

Exemplo de validação por regex:

1. /^\d{1,2}\/\d{1,2}\/\d{4}$/.test(date) => XX/XX/XXXX
2. /^\d{1,2}\-\d{1,2}\-\d{4}$/.test(date) => XX-XX-XXXX
3. /^\d{4}\-\d{1,2}\-\d{1,2}$/.test(date) => XXXX-XX-XX